internal/tray-agent/agent/client/agent-client.go

```diff
@@ -44,11 +53,17 @@ func DestroyMockedAgentController() {
 
 //AgentController is the data structure that manages Tray Agent interaction with the cluster.
 type AgentController struct {
-	client    *crdClient.CRDClient
-	valid     bool
-	connected bool
-	advCache  *AdvertisementCache
-	mocked    bool
+	kubeClient kubernetes.Interface
```

Comment:
it seems to me that this `kubeClient` is connected to the same cluster as `client` field: can't you reuse this one?

Reply:
The inner structure of AgentController needs to be updated in future versions in order to manage also the clients of other CRDs. I preferred separating the two clients at least for the moment. In the future I could consider to use the inner client of the ClusterConfig CRD.